### PR TITLE
Update eks-log-collector.sh

### PR DESCRIPTION
**Pull request by JasonSwindle**
Pretty big re-write.  More sane defaults, tighter logic, systemd / package manager based checks.

### CODE REVIEW

File: eks-log-collector.sh [hunk 1]
```diff
@@ -67,21 +84,23 @@ parse_options() {
...
-ok() {
-  echo "ok"
+ok()
+{
+  echo
 }
 
-info() {
+info()
+{
```
Comment (nithu0115) [inline eks-log-collector.sh]: For uniformity, lets leave this info() {

File: eks-log-collector.sh [hunk 2]
```diff
@@ -116,63 +158,39 @@ is_diskfull()
 {
   try "check disk space usage"
 
-  threshold=70
-  i=2
-  result=`df -kh |grep -v "Filesystem" | awk '{ print $5 }' | sed 's/%//g'`
+  local threshold
+  local result
+  threshold=1500000
+  result=$(df / | grep -v "Filesystem" | awk '{ print $4 }')
```
Comment (nithu0115) [inline eks-log-collector.sh:157]: Can you add h flag to make it human readable instead of bytes?

File: eks-log-collector.sh [hunk 3]
```diff
@@ -190,72 +208,29 @@ pack()
...
-      ;;
-    *)
-      fail
-      die "Unsupported OS detected."
-      ;;
-  esac
+  if [[ -z "${TAR_BIN}" ]]; then
+      warning "TAR archiver not found, please install a TAR archiver to create the collection archive. You can still view the logs in the collect folder."
+    else
+      ${TAR_BIN} --create --verbose --gzip --file "${HOME}"/ekslogsbundle_"${INSTANCE_ID}"_"$(date --utc +%Y-%m-%d_%H%M-%Z)".tar.gz --directory="${COLLECT_DIR}" . > /dev/null 2>&1
```
Comment (nithu0115) [inline eks-log-collector.sh]: Can we have a shorter name ? Something like instanceid_eks_log.tar.gz

File: eks-log-collector.sh (hunk 1 above)
Comment (JasonSwindle) [reply eks-log-collector.sh]: Easy fix.  I do see there are many others where it does not follow ```() {```.  Do you want me to fix these as well?

File: eks-log-collector.sh (hunk 2 above)
Comment (JasonSwindle) [reply eks-log-collector.sh:157]: Later down, I tell the user the space needed:

```
$((threshold>>10))MB
```

As it stands, that is roughly 1500MB or 1.5GB.

File: eks-log-collector.sh (hunk 3 above)
Comment (JasonSwindle) [reply eks-log-collector.sh]: I added date, as there are times where the customer will re-run the log collect script a few times.  If we collect n+1 number of copies of the tar file, they all will be named:

instanceid_eks_log.tar.gz

As it stands, they will be named:
```
ekslogsbundle_i-05922013bdd159670_2018-09-10_2132-UTC.tar.gz
ekslogsbundle_i-05922013bdd159670_2018-09-10_2158-UTC.tar.gz
ekslogsbundle_i-05922013bdd159670_2018-09-10_2159-UTC.tar.gz
```

Let me work on making this shorter.

File: eks-log-collector.sh (hunk 1 above)
Comment (JasonSwindle) [reply eks-log-collector.sh]: Fixed it all, and make  them uniform.

File: eks-log-collector.sh (hunk 3 above)
Comment (JasonSwindle) [reply eks-log-collector.sh]: How does:

```
eks_i-05922013bdd159670_2018-09-10_2159-UTC.tar.gz
```

Sound.  We still know:

- The service
- The instance ID
- Year, Month, Day, Time, Timezone when bundle was made.

Comment (nithu0115) [reply eks-log-collector.sh]: Sounds good.

File: eks-log-collector.sh (hunk 2 above)
Comment (nithu0115) [reply eks-log-collector.sh:157]: Thanks for clarification.

**Comment by nithu0115** [comment]
LGTM!

@JasonSwindle are you going to squash these or would you like me to do so in the merge?

**Comment by nithu0115** [comment]
Thanks @JasonSwindle  👍

**Pull request merged**